src/rpc/connection/dispatch.c

```diff
@@ -198,14 +198,15 @@ int handle_run(connection_request_event_info *info)
     return (-1);
   }
 
-  if (!meta->obj[0].data.string.str || meta->obj[0].data.string.length != 16) {
+  if (!meta->obj[0].data.string.str ||
+    meta->obj[0].data.string.length+1 != PLUGINKEY_STRING_SIZE) {
     error_set(api_error, API_ERROR_TYPE_VALIDATION,
         "Error dispatching run API request. Invalid meta params size");
     return (-1);
   }
 
-  targetpluginkey = meta->obj[0].data.string;
-  string_to_upper(targetpluginkey);
+  memcpy(targetpluginkey, meta->obj[0].data.string.str, PLUGINKEY_STRING_SIZE);
```

Comment:
just use a pointer to `meta->obj[0].data.string.str`. you don't need this memcpy.

```c
char * targetpluginkey = meta->obj[0].data.string.str
```